### PR TITLE
Added attack scenarios and mitigations

### PR DESCRIPTION
**Pull request by yoavweiss**
Hey @vkrasnov! :)

I added a few words on different attack scenarios and mitigations as I see them. I'm not sure the third point is an actual valid point, but wanted to bring it up and verify with sec folks.

It's too early to merge this before such verification takes place, but wanted you to review this as well, so PRing...

### CODE REVIEW

**Comment by yoavweiss** [comment]
Rebased... PTAL

File: draft-vkrasnov-h2-compression-dictionaries.md [hunk 1]
```diff
@@ -103,10 +103,73 @@ document are to be interpreted as described in RFC 2119 {{RFC2119}}.
...
-potentialy sensitive information. We will collaborate with industry experts to
-identify any additional attack vectors introduced by this draft, and include
-a set of best practices to both servers and clients that would implement it.
+The use of compression over an encrypted connection could be used by malicious
+actors to potentially leak sensitive information. We will collaborate with
+industry experts to identify any additional attack vectors introduced by this
+draft, and include a set of best practices to both servers and clients that
+would implement it.
+
+### Attack scenarios and mitigations
```
Comment (vkrasnov) [inline draft-vkrasnov-h2-compression-dictionaries.md]: Not entire sure if it belongs at the beginning, or the end of the document.

File: draft-vkrasnov-h2-compression-dictionaries.md [hunk 2]
```diff
@@ -103,10 +103,73 @@ document are to be interpreted as described in RFC 2119 {{RFC2119}}.
...
+otherwise)
+* CSRF tokens
+* CSP nonces
+* Application level secrets (e.g. financial information, stored credit cards
+numbers, codes, etc.)
+
+The mechanism for such data theft can happen if the attacker can:
+* Download multiple similar payloads to the target page modulo the actual
+secret, while trying out multiple permutations of the secret.
+* Observe the on-the-wire transfer size using Resource Timing's
```
Comment (vkrasnov) [inline draft-vkrasnov-h2-compression-dictionaries.md]: What about passive network monitoring?

File: draft-vkrasnov-h2-compression-dictionaries.md [hunk 3]
```diff
@@ -103,10 +103,73 @@ document are to be interpreted as described in RFC 2119 {{RFC2119}}.
...
+`transferSize` property.
+
+The rest of this section will describe different scenarios where those
+conditions are met as well as potential mitigations for them.
+
+#### Cross-origin secret leak
+An HTTP/2 session can be used to deliver resources from multiple origins over
+which the session has proved to be authoritative. As a result, sharing
```
Comment (vkrasnov) [inline draft-vkrasnov-h2-compression-dictionaries.md]: (connection coalescing)

**Review by vkrasnov — APPROVED** [review]
LGTM overall

File: draft-vkrasnov-h2-compression-dictionaries.md (hunk 1 above)
Comment (yoavweiss) [reply draft-vkrasnov-h2-compression-dictionaries.md]: I'm fine with either

File: draft-vkrasnov-h2-compression-dictionaries.md (hunk 2 above)
Comment (yoavweiss) [reply draft-vkrasnov-h2-compression-dictionaries.md]: I don't think it's a threat here, but I could be wrong

File: draft-vkrasnov-h2-compression-dictionaries.md (hunk 1 above)
Comment (vkrasnov) [reply draft-vkrasnov-h2-compression-dictionaries.md]: Let's keep it where it is for now

**Comment by vkrasnov** [comment]
LGTM

**Pull request merged**